src/Common/ExtraType.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Common;
```

Comment:
I am not sure about the namespace
since more and more will be moved to things like this I think it is important to use a good structure from the start, since changing it can be non BC